src/Microsoft.OpenApi/Extensions/OpenApiTypeMapper.cs

```diff
@@ -152,21 +185,23 @@ public static Type MapOpenApiPrimitiveTypeToSimpleType(this OpenApiSchema schema
             {
                 throw new ArgumentNullException(nameof(schema));
             }
+            var isNullable = (schema.Type & JsonSchemaType.Null) == JsonSchemaType.Null;
+            var nonNullable = (schema.Type & ~JsonSchemaType.Null)?.FirstIdentifier();
```

Comment:
can we remove the toIdentifier part here and use enum and not string comparisons in the switch?

Reply:
Yeah good call